### DISCUSSION TITLE
Store Icon Packs in a Zip Archive

### DISCUSSION THREAD

**Discussion by AbFarid** [FlorianWoelki/obsidian-icon-folder#209] [Ideas]
Hello!
I love this plugin as it helps me visualize the structure much better. However, the current implementation of storing the icon packs as individual files within the plugin folder has led to slow syncs and an inflated file count for my vault.
I've found myself hostage to using only the packs that I already have, because adding (or removing) one would become a real issue syncing, especially to mobile devices.
I'd like to propose a potential solution to this issue.

#### Pros:
- **Reduced Sync Time and File Count**: A zip file will reduce the number of files, leading to faster sync times.
- **Potential Space Savings**: Hey, it might even save some space?

#### Cons:
- **Access Time**: There may be some overhead in extracting files from a zip archive. Hopefully, it would be possible to read individual files directly to memory.
- **Compatibility**: Changing the file structure may necessitate adjustments in how icons are accessed, which could be a non-trivial task.
- **Migration/Backwards Compatibility**: Transitioning from the old structure to the new one may require attention for existing users. Unless it's implemented as a toggle-able option.

**Comment by FlorianWoelki** [comment]
I was actually thinking a lot about this solution and finally had time to implement a short POC. You can test it yourself in the current `master` branch. I'll also test it myself and will notify you whenever it's released.

**Comment by FlorianWoelki** [comment]
This has been implemented and is now resolved!

**Reply by AbFarid** [reply]
Thank you so much! Sorry that I wasn't here to test the feature. Coffee is on me!

Is there a migration guide? Or the update will delete the individual icons automatically?

**Reply by FlorianWoelki** [reply]
Thank you, Cheers! I think [this issue](https://github.com/FlorianWoelki/obsidian-iconize/issues/224) describes the migration quite well. Generally, if you want to integrate the new syncing mechanism, you need to create a zip of your icons and then delete all the unused icons. Or you simply redownload the predefined icon packs (that will create a zip) and readd all of your icons.
